src/main/java/com/epam/esm/service/GiftCertificateService.java

```diff
@@ -53,17 +59,16 @@ public GiftCertificateDto getGiftCertificate(long id) throws EntityNotExistsExce
         return dtoMapper.map(optionalGiftCertificate.get());
     }
 
-    @Transactional
     public void deleteGiftCertificate(long id) throws EntityNotExistsException {
         if (giftCertificateDao.findById(id).isEmpty()) {
             throw new EntityNotExistsException();
         }
 
         giftCertificateTagDao.deleteGiftCertificateById(id);
+        orderDao.deleteByGiftCertificateId(id);
```

Comment:
Why do you delete order when a certificate is deleted? Does it make sense? Is it user-friendly?